include/nccl_ofi_topo.h

```diff
@@ -310,4 +312,49 @@ struct fi_info *nccl_ofi_topo_next_info_list(nccl_ofi_topo_data_iterator_t *iter
  */
 int nccl_ofi_topo_write_nccl_topology(nccl_ofi_topo_t *topo, FILE *file);
 
+class TopologyManager {
```

Comment:
I strongly object to half-refactoring this code.  There's also no need for a topology manager object.